datasets/wmt_mlqe_task3/wmt_mlqe_task3.py

```diff
@@ -0,0 +1,252 @@
+                        "category": datasets.ClassLabel(names=_ANNOTATION_CATEGORIES),
+                    }
+                ),
+                "token_annotations": datasets.Sequence(
+                    {
+                        "segment_id": datasets.Sequence(datasets.Value("int32")),
+                        "first_token": datasets.Sequence(datasets.Value("int32")),
+                        "last_token": datasets.Sequence(datasets.Value("int32")),
+                        "token_after_gap": datasets.Sequence(datasets.Value("int32")),
+                        "severity": datasets.ClassLabel(names=["minor", "major", "critical"]),
+                        "category": datasets.ClassLabel(names=_ANNOTATION_CATEGORIES),
+                    }
+                ),
+                "token_index": datasets.Sequence(datasets.Sequence(datasets.Sequence(datasets.Value("int32")))),
+                "total_words": datasets.Value("int32"),
+            }
+        )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URLs)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir["train+dev"], "task3", "train"),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir["test"], "test-blind"),
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir["train+dev"], "task3", "dev"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+
+        def open_and_read(fp):
+            with open(fp, encoding="utf-8") as f:
+                return f.read().splitlines()
+
+        for id_, folder in enumerate(glob.glob(f"{filepath}/*")):
+            source_segments = open_and_read(os.path.join(folder, "source.segments"))
+            source_tokenized = open_and_read(os.path.join(folder, "source.tokenized"))
+            mt_segments = open_and_read(os.path.join(folder, "mt.segments"))
+            mt_tokenized = open_and_read(os.path.join(folder, "mt.tokenized"))
+
+            if split in ["train", "dev"] and not os.path.exists(os.path.join(folder, "token_index")):
+                logging.warning(
```

Comment:
maybe just mention in the dataset card that some token_index are missing ?
we don't want to pollute the users' logs